fastchat/serve/gradio_block_arena_vision.py

```diff
@@ -4,39 +4,130 @@
 Usage:
 python3 -m fastchat.serve.controller
 python3 -m fastchat.serve.sglang_worker --model-path liuhaotian/llava-v1.5-7b --tokenizer-path llava-hf/llava-1.5-7b-hf
-python3 -m fastchat.serve.gradio_web_server_multi --share --multimodal
+python3 -m fastchat.serve.gradio_web_server_multi --share --vision-arena
 """
 
 import json
 import os
+import time
 
 import gradio as gr
+from gradio.data_classes import FileData
 import numpy as np
 
+from fastchat.constants import (
+    MODERATION_MSG,
+    CONVERSATION_LIMIT_MSG,
+    INPUT_CHAR_LEN_LIMIT,
+    CONVERSATION_TURN_LIMIT,
+)
 from fastchat.serve.gradio_web_server import (
-    upvote_last_response,
-    downvote_last_response,
-    flag_last_response,
     get_model_description_md,
     acknowledgment_md,
     bot_response,
-    add_text,
-    clear_history,
-    regenerate,
     get_ip,
     disable_btn,
+    State,
+    _prepare_text_with_image,
+    get_conv_log_filename,
+    get_remote_logger,
 )
 from fastchat.utils import (
     build_logger,
+    moderation_filter,
 )
 
 logger = build_logger("gradio_web_server_multi", "gradio_web_server_multi.log")
 
+no_change_btn = gr.Button()
+enable_btn = gr.Button(interactive=True, visible=True)
+disable_btn = gr.Button(interactive=False)
+invisible_btn = gr.Button(interactive=False, visible=False)
+visible_image_column = gr.Image(visible=True)
+invisible_image_column = gr.Image(visible=False)
+
 
 def get_vqa_sample():
     random_sample = np.random.choice(vqa_samples)
     question, path = random_sample["question"], random_sample["path"]
-    return question, path
+    res = {"text": "", "files": [path]}
+    return (res, path)
+
+
+def set_visible_image(textbox):
+    images = textbox["files"]
+    if len(images) == 0:
+        return invisible_image_column
+    elif len(images) > 1:
+        gr.Warning(
+            "You can only submit one image for the conversation. Your first image will be used."
```

Comment:
"We only support single image conversations. Please start a new round if you would like to chat using this image."